### PR TITLE
Feature/template varmojis

### PR DESCRIPTION
**Pull request by 121watts**
- [ ] CHANGELOG.md updated
  - [ ] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1262 

### Feature
Much like slack emoji selection, when a user types `:$` into the `QueryMaker` they will see a list of tempVars that they can select with ArrowKeys or the mouse:

### Note
I understand there are mucho merge conflicts with `feature/template-variables`.  Will be waiting to resolve those until this gets the 👍

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,9 +16,13 @@
   1. [#1332](https://github.com/influxdata/chronograf/pull/1332): Remove table toggle from dashboard visualization
 
 ### Features
+  1. [#1292](https://github.com/influxdata/chronograf/pull/1292): Introduce Template Variable Manager
```
Comment (goller) [inline CHANGELOG.md]: All these change log items should be in the next release's section

**Comment by lukevmorris** [comment]
@121watts Just wanted to clarify -- did we end up settling on a simple `:tempVar:` syntax, leaving out any `$`?

**Comment by lukevmorris** [comment]
@121watts says yes

File: ui/src/data_explorer/components/QueryEditor.js [hunk 2]
```diff
@@ -69,55 +198,102 @@ const QueryEditor = React.createClass({
...
-        <Dropdown
-          items={QUERY_TEMPLATES}
-          selected={'Query Templates'}
-          onChoose={this.handleChooseTemplate}
-          className="query-editor--templates"
-        />
+        <div
+          className={classNames('varmoji', {'varmoji-rotated': isTemplating})}
+        >
+          <div className="varmoji-container">
```
Comment (lukevmorris) [inline ui/src/data_explorer/components/QueryEditor.js]: I wonder if it's incorrect to attach the `varmoji` term this far up the tree -- `varmoji-container` contains both varmoji behavior as well as indicating query status, which is a separate concept.

It looks like we already use the term `drawer` to refer to this area, i.e. opening/closing the drawer, controls the value of `isTemplating`. Maybe use `drawer` or a variant thereof for the namespace here?

**Comment by lukevmorris** [comment]
I reviewed this PR mostly locally with `git diff feature/template-variables..feature/template-varmojis` to cut out as much noise as possible.

This looks good for now. There seems to be quite a bit of complex template variable logic in QueryMaker; we should find a conceptual home for that logic so the component can be slimmed down as much as possible to just render logic.

Given the garbled state of this branch, I recommend the following merge strategy:
- Merge `feature/template-variables` => `feature/template-varmojis`
- Resolve merge conflicts
- Do a final review pass to double check the actual changeset
- Merge this PR

File: LICENSE_OF_DEPENDENCIES.md [hunk 3]
```diff
@@ -676,7 +676,7 @@
 * node-libs-browser 0.6.0 [MIT](https://github.com/webpack/node-libs-browser)
 * node-notifier 4.6.1 [MIT](ssh://git@github.com/mikaelbr/node-notifier)
 * node-pre-gyp 0.6.29 [BSD-3-Clause](http://github.com/mapbox/node-pre-gyp)
-* node-sass 3.11.3 [MIT](https://github.com/sass/node-sass)
+* node-sass 4.5.2 [MIT](https://github.com/sass/node-sass)
```
Comment (121watts) [inline LICENSE_OF_DEPENDENCIES.md]: yarn

**Pull request merged**